editoast/src/views/timetable.rs

```diff
@@ -338,14 +339,29 @@ async fn conflicts(
         );
     }
     let conflict_detection_request = ConflictDetectionRequest {
+        infra: infra_id,
+        expected_version: infra.version,
         trains_requirements,
-        infra_id,
+        work_schedules: None,
     };
 
     // 3. Call core
-    let conflicts = conflict_detection_request.fetch(&core_client).await?;
+    let conflict_detection_response = conflict_detection_request.fetch(&core_client).await?;
 
-    Ok(Json(conflicts.conflicts))
+    // 4. Remove work_schedule conflicts
+    let conflicts = conflict_detection_response
+        .conflicts
+        .into_iter()
+        .filter_map(|conflict_with_work_schedules| {
+            if conflict_with_work_schedules.conflict.conflict_type != ConflictType::WorkSchedule {
+                Some(conflict_with_work_schedules.conflict)
+            } else {
+                None
+            }
+        })
+        .collect();
```

Comment:
I'm really not sure that we should be filtering in this endpoint. Work schedules are associated with an infra, this endpoint takes a timetable as an input, which is itself associated with an infra: semantically, this endpoint should return work schedule conflicts as well. Only the `front` knows the context in which it will be used and has the ability to decide if work schedule conflicts should be displayed or not. What do you think?

Reply:
I'll let @flomonster answer this, but the reasoning was that it was for sure not planned to use this endpoint directly from the frontend with work_schedules, which is why we filter it in post-processing. Personally don't mind either way, adds very few modifications.

Reply:
> After further talk, this should also mean that there shouldn't be any distinction in the conflict type bewteen a spacing conflict and a work schedule conflict. I'm down to remove the distinction (i'll do it in a separate commit to revert in case you don't agree, but i think it'll be better).

This should also resolve this problem :)

Reply:
Actually we provide an empty list of work schedule. Having workschedule conflict is completly unexpected here. We should either remove this filter or assert core response :shrug: